src/components/reusable/progressBar/progressBar.ts

```diff
@@ -0,0 +1,273 @@
+import { html, LitElement } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+import '@kyndryl-design-system/shidoka-foundation/components/icon';
+import '../loaders';
+import '../tooltip';
+import checkmarkIcon from '@carbon/icons/es/checkmark--filled/16';
+import errorIcon from '@carbon/icons/es/error--filled/16';
+import informationIcon from '@carbon/icons/es/information/16';
+
+import ProgressBarStyles from './progressBar.scss';
+
+enum ProgressStatus {
+  INDETERMINATE = 'indeterminate',
+  ACTIVE = 'active',
+  SUCCESS = 'success',
+  ERROR = 'error',
+}
+
+/**
+ * `<kyn-progress-bar>` -- progress bar status indicator component.
+ */
+@customElement('kyn-progress-bar')
+export class ProgressBar extends LitElement {
+  static override styles = ProgressBarStyles;
+
+  /** Sets visibility of optional inline load status spinner. */
+  @property({ type: Boolean })
+  showInlineLoadStatus = true;
+
+  /** Sets progress bar status mode. */
+  @property({ type: String })
+  status: 'indeterminate' | 'active' | 'success' | 'error' = 'indeterminate';
```

Comment:
Does 'indeterminate' need to be a state, or can it be derived from absence of `value`?